### PR TITLE
GraphQL (Docs): Document the /admin/schema/validate endpoint

### PR DESCRIPTION
**Pull request by aaroncarey**
Added documentation for the schema validation endpoint, per https://github.com/dgraph-io/dgraph/pull/6250 and GRAPHQL-578.

Fixes GRAPHQL-780

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6814)

 
 

Docs Preview: [Dgraph Preview](https://dgraph-3d4735a862-105490.surge.sh)

### CODE REVIEW

**Review by abhimanyusinghgaur — CHANGES_REQUESTED** [review]
Suggested some minor wording changes and a few spelling mistakes.
:lgtm: after those are addressed.

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-:-MKsJPBA9XIc1gqWVjtf:bhqwh2d)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @aaroncarey, @abhimanyusinghgaur, @bucanero, @danielmai, and @MichaelJCompton)

___
*[wiki/content/graphql/admin/index.md, line 348 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsJBHp7Aft9SO0MXdE:-MKsJBHp7Aft9SO0MXdF:b-9naugm) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L348)):*
> ```
> defnition 
> ```
definition
___
*[wiki/content/graphql/admin/index.md, line 349 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsIiHl30Blu88QyTp8:-MKsIiHmDU8cmg2d4cr4:b-qz9en2) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L349)):*
<details><summary><i alt="summary">Quoted 4 lines of code…</i></summary>

> ```
> You can validate a GraphQL schema before adding it to your database by sending
> your schema defnition to the `/admin/schema/validate` endpoint, as in the
> following HTTP request example:
> Request header:
> ```
</details>
ssh
> path: /admin/schema/validate
> method: POST
> content-type: application/json
> ```
> Request body:
> ```
Should we phrase it like this:
```
You can validate a GraphQL schema before adding it to your database by sending
your schema definition in an HTTP POST request to the `/admin/schema/validate` endpoint, with the following body:

```
___
*[wiki/content/graphql/admin/index.md, line 356 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsHv9l7raSs4q7O4ct:-MKsHv9l7raSs4q7O4cu:b-xhq5wc) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L356)):*
> ```
> content-type: application/json
> ```
This header is not required in the HTTP request. The input is just plain text.
The HTTP response is JSON.
___
*[wiki/content/graphql/admin/index.md, line 368 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsIm-D9PH_Vi3kk-5S:-MKsIm-D9PH_Vi3kk-5T:b-k00oht) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L368)):*
> ```
> and
> ```
an

**Review by abhimanyusinghgaur — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-:-MKsJ_xk8BufiMuC77RR:bnfp4nl)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @aaroncarey, @abhimanyusinghgaur, @bucanero, @danielmai, and @MichaelJCompton)

**Review by aaroncarey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6814)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @abhimanyusinghgaur, @bucanero, @danielmai, and @MichaelJCompton)

___
*[wiki/content/graphql/admin/index.md, line 348 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsJBHp7Aft9SO0MXdE:-MKuNHNU7005p8BtzQsS:b-tem3sn) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L348)):*
<details><summary><i>Previously, abhimanyusinghgaur (Abhimanyu Singh Gaur) wrote…</i></summary><blockquote>

> ```
> defnition 
> ```
definition
</blockquote></details>

Done. Sorry that I missed that!
___
*[wiki/content/graphql/admin/index.md, line 349 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsIiHl30Blu88QyTp8:-MKuNr2EBXDirrZKeFL3:bxmcggm) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L349)):*
<details><summary><i>Previously, abhimanyusinghgaur (Abhimanyu Singh Gaur) wrote…</i></summary><blockquote>

> ```
> You can validate a GraphQL schema before adding it to your database by sending
> your schema defnition to the `/admin/schema/validate` endpoint, as in the
> following HTTP request example:
> Request header:
> ```ssh
> path: /admin/schema/validate
> method: POST
> content-type: application/json
> ```
> Request body:
> ```
Should we phrase it like this:
```
You can validate a GraphQL schema before adding it to your database by sending
your schema definition in an HTTP POST request to the `/admin/schema/validate` endpoint, with the following body:

```
</blockquote></details>

Thanks for this suggestion! I picked up some of this phrasing, but I wanted to retain the word "example" and also provide the example header that includes the `content-type`, as including that in the sentence made it too long.
___
*[wiki/content/graphql/admin/index.md, line 356 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsHv9l7raSs4q7O4ct:-MKuP4VH9eYG5jUE3ySO:b-1y8lq3) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L356)):*
<details><summary><i>Previously, abhimanyusinghgaur (Abhimanyu Singh Gaur) wrote…</i></summary><blockquote>

> ```
> content-type: application/json
> ```
This header is not required in the HTTP request. The input is just plain text.
The HTTP response is JSON.
</blockquote></details>

Thanks for the clarification! Fixed.
___
*[wiki/content/graphql/admin/index.md, line 368 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-MKsIm-D9PH_Vi3kk-5S:-MKuPDmr8nGi_aIYXI8J:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/a179e80d05cb1fac6e0c0e854f74cc2e7ac56951/wiki/content/graphql/admin/index.md#L368)):*
<details><summary><i>Previously, abhimanyusinghgaur (Abhimanyu Singh Gaur) wrote…</i></summary><blockquote>

> ```
> and
> ```
an
</blockquote></details>

Done.

File: wiki/content/graphql/admin/index.md [hunk 1]
```diff
@@ -342,11 +342,36 @@ On first starting with a blank database:
...
+```graphql
+type Person {
+	name: String
+}
+```
+
+This endpoint returns a JSON response that indicates if the schema is valid or
+not, and provides an error if isn't valid. In this case, the schema is valid,
+so the JSON response includes the following message: `Schema is valid`.
+
```
Comment (bucanero) [inline wiki/content/graphql/admin/index.md:369]: if you want, we could also add a `curl` example, like:

```sh
curl -X POST localhost:8080/admin/schema/validate --data-binary '@schema.graphql'
```

**Review by bucanero — COMMENTED** [review]
LGTM

**Review by aaroncarey — COMMENTED** [review]
Thanks!

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6814#-:-MKvcxgd2ybTmEJR-ls4:b-ucfknh)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @aaroncarey, @abhimanyusinghgaur, @danielmai, and @MichaelJCompton)

**Pull request merged**